common/thorhelper/thorstore.hpp

```diff
@@ -0,0 +1,170 @@
+ *    - passing #parts that it will be read from to allow split information to be optimized.
+ *    A) thor
+         - Translate logical nodes to physical ips.
+         - gather any missing file sizes
+      B) s3/azure blobs
+         - Expand any wildcards in the filenames and create entries for each expanded file.
+         - gather file sizes
+      C) HDFS
+         - gather files sizes
+         - gather split points
+ * c) serialize to the slaves
+ * d) deserialize from the master
+ * e) call fileCollection::partition(numChannels, myChannel) to get a list of partitions
+ * f) iterate though the partitions for the current channel
+ *    a) need a reader for the format - how determine?
+ *    b) Where we determine if it should be read directly or via dafilesrv?
+ *    c) request a row reader for the
+ *
+ * Questions:
+ *    Where are the translators created?
+ *    What is signed?  It needs to be a self contained block of data that can easily be serialized and deserialized.
+ *      I don't think it needs to contain information about the storage array - only the logical file.
+ */
+
+class CStorageSystems;
+
+//What is a good term for a collection of drives
+//storage array/system/
+
+//This describes the a set of disks that can be used to store a logical file.
+//  "device" is used to represent the index within the storage plane
+class THORHELPER_API CStorageHostGroup : public CInterface
+{
+public:
+    CStorageHostGroup(const IPropertyTree * _xml);
+
+    const char * queryName() const;
+    const char * queryHost(unsigned idx) const;
+    bool isLocal(unsigned device) const;
+
+private:
+    const IPropertyTree * xml;
+};
+
+
+//This describes the a set of disks that can be used to store a logical file.
+//  "device" is used to represent the index within the storage plane
+class THORHELPER_API CStoragePlane : public CInterface
+{
+    friend class CStorageSystems;
+
+public:
+    CStoragePlane() = default;
+    CStoragePlane(const IPropertyTree * _xml, const CStorageHostGroup * _host);
+
+    bool containsHost(const char * host) const;
+    bool containsPath(const char * path);
+    bool matches(const char * search) const { return strsame(name, search); }
+    bool matchesHost(const char * host);
+
+//    unsigned getCost(unsigned device, const IpAddress & accessIp, PhysicalGroup & peerIPs) const;
+    StringBuffer & getURL(StringBuffer & target, unsigned part) const;
+    unsigned getWidth() const;
+    bool isLocal(unsigned part) const;
+    bool isAttachedStorage() const;
+    const char * queryName() const { return name; }
+    const char * queryPath() const;
+    const char * queryScopeSeparator() const;
+
+protected:
+    unsigned getNumDrives() const { return 1; }  // MORE: Should we allow support for drives as well as devices?
+    unsigned getDevice(unsigned part) const;
+    unsigned getDrive(unsigned part) const;
+
+private:
+    const IPropertyTree * xml;
+    const char * name;
```

Comment:
set to nullptr for consistency? [even though (like others) it is set in the ctor]